### PR TITLE
fix(inference): strip falsy logprobs from Ollama request payloads

### PR DESCRIPTION
**Pull request by Ahmath-Gadji**
## Problem

`OllamaClient` builds every request body by merging two sources:

```python
payload = {**self._defaults, **kwargs, "model": ..., "messages": ..., "stream": False}
```

`self._defaults` carries the endpoint config, and `LLMParamsConfig.logprobs` defaults to `False` (`conf/config.yaml`, `core/config/endpoints.py`). So `logprobs: false` is merged into **every** chat, stream and completion request — even when the client sent nothing.

Ollama's OpenAI-compatible `/v1` API does not support `logprobs`; it is only implemented on the native `/api/chat` and `/api/generate` endpoints.

Reproduced with a mock transport, with no client-supplied `logprobs` at all:

```
{"logprobs": false, "temperature": 0.1, "model": "m", "messages": [...], "stream": false}
```

## Fix

`VLLMClient` already guards against exactly this — `_strip_falsy_logprobs()` exists and its docstring names this scenario: *"the config default (`LLMParamsConfig.logprobs = False`) lands in `self._defaults` and is sent on every request."* The helper was simply never wired into `OllamaClient`.

Reuse it in `chat`, `stream_chat` and `generate`. `ollama_client` already imports `_parse_response` from `vllm_client`, so this needs no new module — 1 import line and 3 calls.

## Semantics preserved

Only `None` and `False` are stripped, via `is` comparisons:

- a truthy `logprobs` is a deliberate opt-in and is forwarded with its `top_logprobs`
- `/v1/completions` takes `logprobs` as an **integer** count, where `0` means "the sampled token's own logprob, no alternates" — a real request, not an off state. It survives.

## Testing

6 new tests in `TestOllamaClient`. Verified they are not vacuous: disabling the three strip calls fails the 4 stripping tests, while the 2 pass-through tests stay green (they guard the opposite direction — over-stripping).

`uv run pytest tests/unit/services/inference` → 294 passed.

## Note

Found by CodeRabbit on #897. That PR does not cause this — it was already happening on `develop` from the config default alone — so the fix is split out here.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved inference request handling by filtering empty or false `logprobs` values before sending requests.
  * Preserved valid `logprobs` configurations, including values used to request response details.

* **Tests**
  * Added coverage for chat, streaming, and generation requests with empty, false, zero, and configured `logprobs` values.

### CODE REVIEW

**Review by hedhoud — DISMISSED** [review]
Reviewed the request-payload behavior and its edge cases. The change prevents unsupported falsy logprobs from reaching Ollama while preserving explicit values. Focused tests, lint, formatting, and all CI checks pass.

File: openrag/services/inference/ollama_client.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 from ._call_log import log_llm_call
 from ._circuit_breaker import with_circuit_breaker
 from ._retry import with_retry
-from .vllm_client import _parse_response
+from .vllm_client import _parse_response, _strip_falsy_logprobs
```
[the remark this answers is not in the document]
Comment (Ahmath-Gadji) [reply openrag/services/inference/ollama_client.py]: Not applying this one — the relative import is the convention here, and the proposed `openrag.`-prefixed form is the style this repo deliberately moved away from.

**1. `CLAUDE.md` scopes the absolute-import rule to *cross-package* imports and explicitly permits relative ones inside a package:**

```python
# Avoid relative imports across packages
# from .ray_utils import ...  # Only within same package
```

`ollama_client.py` and `vllm_client.py` are both in `openrag/services/inference/`, so this is a same-package sibling import. Lines 29–31 directly above it (`._call_log`, `._circuit_breaker`, `._retry`) and `services/inference/__init__.py` use the same form. This PR only appended `_strip_falsy_logprobs` to an import that was already there — the line's style is pre-existing and unchanged.

**2. The suggested prefix is not this project's absolute style.**

The Python path root is `openrag/` (`pythonpath = ["./openrag"]` in `pyproject.toml`), so canonical absolute imports read `from core.X` / `from services.X`. There are 346 of those in the package against 21 `from openrag.X`. PR #913 (`fix/885-canonical-imports`) rewrote `from openrag.core.X` → `from core.X` specifically to strip that prefix, and this very file already uses `from core.utils.logging import get_logger` two lines up.

Applying the suggestion verbatim would reintroduce the prefix #913 removed and leave four sibling imports in a single block written in two different styles.

Both enforced guards agree — `layer-import-guard` and `lint (3.12)` pass on this branch.

**Review by hedhoud — APPROVED** [review]
